lib/doorkeeper/grant_flow.rb

```diff
@@ -0,0 +1,44 @@
+require 'doorkeeper/grant_flow/flow'
+
+module Doorkeeper
+  module GrantFlow
+    mattr_accessor :registered_flows
+    self.registered_flows = {}
+
+    module_function
+
+    def register(name_or_flow, options = {})
+      if name_or_flow.is_a? Doorkeeper::GrantFlow::Flow
+        registered_flows[name_or_flow.name.to_sym] = name_or_flow
+      else
+        registered_flows[name_or_flow.to_sym] = Flow.new(name_or_flow, options)
+      end
+    end
+
+    def get(name)
+      registered_flows.fetch(name.to_sym)
+    end
+  end
+end
+
+Doorkeeper::GrantFlow.register :implicit,
+  response_type_matches: 'token',
```

Comment:
Aligning these here would start to push up against line-length limits. I may poke around with Hound in my own repo to see if there's a readable format that it likes…